src/Integration.UnitTests/LocalServices/RuleSetInspectorTests.cs

```diff
@@ -0,0 +1,494 @@
+            ((IDisposable)this.temporaryFiles).Dispose();
+        }
+        #endregion
+
+        #region Properties
+        /// <summary>
+        /// Simulates the solution level SonarQube folder in which we store the fetched rulesets
+        /// </summary>
+        private string SonarQubeRuleSetFolder
+        {
+            get
+            {
+                return this.TestContext.TestRunDirectory;
+            }
+        }
+
+        /// <summary>
+        /// Simulates the usages of the vs ruleset folder for the default rulesets that ship with VS
+        /// </summary>
+        private string VsRuleSetsDirectory
+        {
+            get
+            {
+                return Path.Combine(this.TestContext.TestRunDirectory, RuleSetInspector.DefaultVSRuleSetsFolder);
+            }
+        }
+        #endregion
+
+        #region Tests
+        [TestMethod]
+        public void RuleSetInspector_FindConflictingRules_ProjectLevelOverridesOfTheSolutionRuleset()
+        {
+            // Setup
+            string solutionRuleSet = this.CreateCommonRuleSet().FilePath;
+            this.temporaryFiles.AddFile(solutionRuleSet, false);
+
+            // Check all supported RuleAction values
+            RuleAction[] unsupportedActions = new[] { RuleAction.Default };
+            foreach (RuleAction ruleAction in Enum.GetValues(typeof(RuleAction)).OfType<RuleAction>().Except(unsupportedActions))
+            {
+                foreach (IncludeType includeType in Enum.GetValues(typeof(IncludeType)).OfType<IncludeType>())
+                {
+                    this.TestContext.WriteLine("Running test case, Project Rules are {0}, SolutionInclude is {1}", ruleAction, includeType);
+
+                    RuleSet projectRuleSet = CreateProjectRuleSetWithInclude(DefaultNumberOfRules, solutionRuleSet, includeType, ruleAction);
+                    this.temporaryFiles.AddFile(projectRuleSet.FilePath, false);
+
+                    // Act
+                    RuleConflictInfo conflicts = this.testSubject.FindConflictingRules(solutionRuleSet, projectRuleSet.FilePath);
+
+                    // Verify
+                    if (ruleAction == RuleAction.None)
+                    {
+                        AssertMissingRulesByFullIds(conflicts, projectRuleSet.Rules);
+                        AssertNoWeakRules(conflicts);
+                    }
+                    else if (ruleAction == RuleAction.Info || ruleAction == RuleAction.Hidden)
+                    {
+                        AssertWeakRulesByFullIds(conflicts, projectRuleSet.Rules);
+                        AssertNoMissingRules(conflicts);
+                    }
+                    else
+                    {
+                        AssertNoConflicts(conflicts);
+                    }
+                }
+            }
+        }
+
+        [TestMethod]
+        public void RuleSetInspector_FindConflictingRules_VsIncludesCannotCreateConflictsByDefault()
```

Comment:
Which is why I added a note at the top of the file...
/* Notes: "By default" is referred to the way we create rulesets for projects when binding.
    ...
*/